src/de/mod.rs

```diff
@@ -370,7 +373,11 @@ impl<'a, 'de> de::Deserializer<'de> for &'a mut Deserializer<'de> {
         match peek {
             b'"' => {
                 self.eat_char();
-                visitor.visit_string(self.parse_string()?)
+                let str_like = self.parse_string()?;
+                match str_like {
+                    StringLike::Borrowed(str) => visitor.visit_borrowed_str(str),
```

Comment:
I just double checked this. `visit_borrowed_str` comes with stronger lifetime guarantees than `visit_str` (lifetime of `str` is `'de`). So this is the correct way to call it.